### ISSUE TITLE
Split out stylistic roles similar to eslint

### ISSUE THREAD

**Issue by slalomzacharyd** [stylelint/stylelint#7362]
### What is the problem you're trying to solve?

As a user, I want to be able to not see errors for rules that are fixable by stylelint.  eslint finally supports this more out of the box with newer versions and stylelint should follow the same pattern.

While stylelint supports manually changing warnings levels, its a bit abusive.  In order to do so, you need to copy all of the rules from either standard or recommened, then individually check which ones are fixable on the website. There is no metadata to find fixable rules which are almost always stylistic.

### What solution would you like to see?

- Either a solution similar to eslint -> https://eslint.style/
- Or at least some kind of supported metadata to programmically find the fixable rules.

A lot of linters have a pattern of making a wall of busy work errors when the majority are auto fixable by the app.  There is no reason to create busy work or try to distract engineers.

**Comment by ybiquitous** [comment]
@slalomzacharyd Thanks for using the template. I understand your motivation well. But your expectation seems to have been realized. See this [online demo on Stackblitz](https://stackblitz.com/edit/stackblitz-starters-hmplvt?file=test.css).

<img width="514" alt="image" src="https://github.com/stylelint/stylelint/assets/473530/058f25e3-710f-47a8-80f9-8c88557074e4">

You expect Stylelint doesn't output anything when using `--fix`, right?

<details><summary>Code</summary>
<p>

Using Stylelint 15.11.0:

```css
@import url('foo.css');
```

```js
module.exports = {
  rules: {
    'import-notation': 'string',
  },
};
```

</p>
</details>

**Comment by jeddy3** [comment]
@slalomzacharyd Thanks for the request.

> I want to be able to not see errors for rules that are fixable by stylelint

As @ybiquitous said, this is current behaviour when `--fix` is used. There's the related https://github.com/stylelint/stylelint/issues/7338 to signpost this better when it's not.

---

Am I right in thinking that the crux of the issue is?:

>  There is no metadata to find fixable rules _which are almost always stylistic_ \[emphasis mine\].

We currently [split our rules into two](https://stylelint.io/user-guide/rules): 

- avoid errors
- enforce conventions

(For reference, ESLint does three: [Possible Problems](https://eslint.org/docs/latest/rules/#possible-problems), [Suggestions](https://eslint.org/docs/latest/rules/#suggestions), and [Layout & Formatting](https://eslint.org/docs/latest/rules/#layout--formatting).

We currently have [meta.fixable](https://stylelint.io/developer-guide/rules#add-autofix), but there is no way to programmatically know what category a rule belongs.

> eslint finally supports this more out of the box with newer versions

Are you referring to the `type` metadata and [`--fix-type`](https://eslint.org/docs/latest/use/command-line-interface#--fix-type) option?

We can discuss adding something similar to Stylelint. If we do, we'll want to lock down the types as they'll become part of our public API rather than just for documentation.
